### PR TITLE
improv: use statusFilter instead of includeArchive,includeDrafts for document search

### PR DESCRIPTION
**Pull request by Pranav2612000**
Refer: https://github.com/outline/outline/discussions/6425

### CODE REVIEW

File: server/routes/api/documents/schema.ts [hunk 1]
```diff
@@ -25,6 +25,10 @@ const DocumentsSortParamsSchema = z.object({
 
 const DateFilterSchema = z.object({
   /** Date filter */
+  statusFilter: z
+    .union([z.literal("published"), z.literal("archived"), z.literal("draft")])
```
Comment (tommoor) [inline server/routes/api/documents/schema.ts:29]: Should be able to use `z.nativeEnum` to share the type definition

File: server/models/helpers/SearchHelper.ts [hunk 2]
```diff
@@ -354,36 +352,51 @@ export default class SearchHelper {
...
             },
+            archivedAt: {
+              [Op.eq]: null,
+            },
           },
+        ],
+      });
+    }
+
+    if (options.statusFilter && options.statusFilter.includes("draft")) {
```
Comment (tommoor) [inline server/models/helpers/SearchHelper.ts:371]:
```suggestion
    if (options.statusFilter?.includes("draft")) {
```

Collapse these clauses

**Review by tommoor — COMMENTED** [review]
Looks reasonable so far, thanks!

File: server/models/helpers/SearchHelper.test.ts [hunk 3]
```diff
@@ -207,48 +206,119 @@ describe("SearchHelper", () => {
...
+      });
       const { results } = await SearchHelper.searchForUser(user, "test", {
-        includeDrafts: false,
+        statusFilter: [StatusFilter.Published],
       });
-      expect(results.length).toBe(0);
+      expect(results.length).toBe(1);
     });
 
-    test("should not include drafts with user permission", async () => {
```
Comment (Pranav2612000) [inline server/models/helpers/SearchHelper.test.ts:230]: @tommoor I'm not sure what this test does and if it still applies. Can you help me with this? Do we still want to keep this condition?

**Comment by Pranav2612000** [comment]
@tommoor Addressed your feedback and updated the tests.  Had some questions about the `membership.id` and `createdById` check we had when search in drafts. ( https://github.com/outline/outline/pull/6537#discussion_r1491894403 )
Do we still need those checks? What do they do?
Thank you.

**Comment by Pranav2612000** [comment]
Also will start with the updates to the UI. I can see the current `FilterOption` component we have for choosing the status only allows choosing one option at a time.
image

Do you have a different component for "FilterOption" which allows choosing multiple items at a time? Or do you want me to build it out?

File: server/models/helpers/SearchHelper.test.ts (hunk 3 above)
Comment (tommoor) [reply server/models/helpers/SearchHelper.test.ts:230]: The test is checking that `includeDrafts=false` filters correctly when there is a user membership to the document. Some equivalent should be easy enough to keep

**Comment by tommoor** [comment]
> Do you have a different component for "FilterOption" which allows choosing multiple items at a time? Or do you want me to build it out?

There isn't anything yet, this is what I was referring to in [this comment](https://github.com/outline/outline/discussions/6425#discussioncomment-8436940) – changing `activeKey` to a plural `activeKeys` would do the trick. Or honestly in hindsight i dislike this naming, could be `selectedItems` or something more clear.

File: server/models/helpers/SearchHelper.test.ts (hunk 3 above)
Comment (Pranav2612000) [reply server/models/helpers/SearchHelper.test.ts:230]: Yup. I've re-added the tests with some minor changes. Can you verify this once when you have time?

**Comment by Pranav2612000** [comment]
Thank you for the reviews @tommoor 
I've done the updates. Can you take another look.

Updated the `FilterOptions` to add support for Multi selecting values

File: server/models/helpers/SearchHelper.ts [hunk 4]
```diff
@@ -354,36 +352,57 @@ export default class SearchHelper {
...
+
+    if (options.statusFilter?.includes(StatusFilter.Archived)) {
+      statusQuery.push({
+        archivedAt: {
           [Op.ne]: null,
         },
       });
     }
 
+    if (options.statusFilter?.length) {
```
Comment (tommoor) [inline server/models/helpers/SearchHelper.ts:400]: Previously archived documents were filtered by default, we should retain this behavior. Given the new changes it probably makes sense to do this on the frontend by making `published,draft` statusFilter enabled in the client by default.

**Comment by Pranav2612000** [comment]
Thank you for the review @tommoor .
I've removed the notes, updated defaultLabel and enabled `published` & `draft` by default.

**Comment by tommoor** [comment]
I will checkout and test properly tomorrow, but I think this is ok. A good follow up issue will be to add the option to also search trashed documents.

**Review by tommoor — APPROVED** [review]

**Comment by tommoor** [comment]
Technically this is a breaking API change but we'll get away with it

**Comment by Pranav2612000** [comment]
Thank you for the opportunity @tommoor and for the final finishing touches.

**Pull request merged**